src/core/annotation.js

```diff
@@ -1251,6 +1255,15 @@ class WidgetAnnotation extends Annotation {
     });
     data.defaultFieldValue = this._decodeFormValue(defaultFieldValue);
 
+    if (fieldValue === undefined && params.xfaDatasets) {
+      // Try to figure out if we have something in the xfa dataset.
+      const path = stringToPDFString(dict.get("T") || "");
```

Comment:
*I don't have time to do a full review right now, but wanted to mention one thing straight away.*

This ought to be available already, via `this._title.str` (note the `setTitle`-call in the `Annotation`- constructor), and you shouldn't need to fetch/parse this again here.